components/motor/gpio/motor_encoder.go

```diff
@@ -529,9 +573,17 @@ func (m *EncodedMotor) GetMotorState(ctx context.Context) EncodedMotorState {
 // Stop stops rpmMonitor and stops the real motor.
 func (m *EncodedMotor) Stop(ctx context.Context, extra map[string]interface{}) error {
 	m.stateMu.Lock()
+	defer m.stateMu.Unlock()
 	m.state.goalRPM = 0
 	m.state.regulated = false
-	m.stateMu.Unlock()
+
+	// after the motor is created, Stop is called, but if the PID controller
+	// is auto-tuning, the loop needs to keep running
+	if m.loop != nil && !m.loop.GetTuning(ctx) {
+		m.loop.Stop()
+		m.loop = nil
```

Comment:
specifically, setting the loop to nil fixed the issue I was having that the control loop wasn't being closed as expected and multiple control loops were conflicting with each other. is it fine to set the loop to nil in this way or is there a better way to go about this?

Reply:
to answer your question, this is fine, it is conceptually the same pattern as calling a cancel function on a context, which will make a context nil I believe.

Reply:
but we should be setting that channel to nil in the control loop if needed.